src/lib.rs

```diff
@@ -212,56 +242,62 @@ impl<O> IO<O> {
 /// assert_eq!(result, "meow\n");
 /// # Ok (()) }
 /// ```
-pub fn pure <T> (value: T) -> IO<T> {
+pub fn pure <'a, T> (value: T) -> IO<'a, T> where T: 'a {
     IO (Pure (value))
 }
 
 /// Ignore the output value of `x`.
-pub fn void <T> (x: IO<T>) -> IO<()>
+pub fn void <'a, T> (x: IO<'a, T>) -> IO<'a>
 where
     T: 'static
 {
     x.replace(())
 }
 
```

Comment:
I feel like the various IO primitives(? I think that's what you call them) belong in a different module. If you wanna get fancy with it I think the rest of the crate could do just fine only relying on `alloc`, so you could make `std` an optional dependency